BootloaderCommonPkg/Library/UefiVariableLib/UefiVariableLib.c

```diff
@@ -692,6 +692,11 @@ GetVariableHeader (
       // Variable header pointed by Variable is inconsecutive,
       //
       LibStatus = GetLibraryData (PcdGet8 (PcdUefiVariableLibId), (VOID **)&VarStoreLibData);
+      if (LibStatus == EFI_NOT_FOUND) {

```

Comment:
If LibStatus == EFI_NOT_FOUND, I think we should just return FALSE ?  
Allocating a new buffer will not work since all critical members in the structure won't be initialized.

Reply:
Also, Status can be anything, other than EFI_SUCCESS in failure case. So please check for EFI_ERROR(Status), instead of only EFI_NOT_FOUND.